api4/user.go

```diff
@@ -3115,15 +3115,7 @@ func setUnreadThreadByPostId(c *Context, w http.ResponseWriter, r *http.Request)
 		return
 	}
 
-	post, err := c.App.GetSinglePost(c.Params.PostId)
-
-	if err != nil {
-		c.Err = err
-		return
-	}
-
-	var thread *model.ThreadResponse
-	thread, err = c.App.UpdateThreadReadForUser(c.AppContext.Session().Id, c.Params.UserId, c.Params.TeamId, c.Params.ThreadId, post.CreateAt-1)
+	thread, err := c.App.UpdateThreadReadForUserByPost(c.AppContext.Session().Id, c.Params.UserId, c.Params.TeamId, c.Params.ThreadId, c.Params.PostId)
```

Comment:
One other thought -- and it's probably not an issue -- but do we need to check that the user has access to the post in question? Thinking this effectively lets me leak the creation timestamp of any post in the system. Any perspective, @jupenur?

Reply:
Good catch @lieut-data! Yes, this needs a permission check. It doesn't just leak the timestamp, but also things like the number of mentions, so someone could use this API to check whether they've been mentioned in threads they don't otherwise have access to.